health/health_json.c

```diff
@@ -142,6 +142,24 @@ static inline void health_rrdcalc2json_nolock(RRDHOST *host, BUFFER *wb, RRDCALC
     char value_string[100 + 1];
     format_value_and_unit(value_string, 100, rc->value, rc->units, -1);
 
+    char *replaced_info = NULL;
+    if (likely(rc->info)) {
+
+        replaced_info = strdupz(rc->info);
+
+        if ( strstr (replaced_info, "$this") ) {
```

Comment:
I will give some suggestions.

In normal situation  the `strstr` implementation is slow, as you can see [here](https://opensource.apple.com/source/tcl/tcl-10/tcl/compat/strstr.c.auto.html), and [here](https://github.com/lattera/freebsd/blob/master/sys/libkern/strstr.c). You are calling this function many times here and also in find_and_replace, what is your opinion about instead to pass as argument the text we pass the pointer and an identification from an `enum`?

Now a doubt, your code is easy to keep with these three variables, but are we going to extend it with more variables? If we need to do this, probably we need to do something like we have for `libnetdata/eval/`.

Best regards,

Thiago

Reply:
Hi Thiago, thank you very much for your input, I will look into it.

Not sure if we're planning for more than three variables...